### DISCUSSION TITLE
A WebGL could not be created - weird issue

### DISCUSSION THREAD

**Discussion by piotrszpruta** [google/model-viewer#4293] [Q&A]
Hello,
I have an issue with some pc's. When I'm viewing model and changing textures on my machine it works well. On mobile also, but i've checked on other 2 computers and textures appear I think darker? 
Photos from my view:
![image](https://github.com/google/model-viewer/assets/38475451/fd01af9c-7074-4871-9b0a-6f1932e70ec1)

Other pc:
![346162405_190374126862622_5371695016863790344_n](https://github.com/google/model-viewer/assets/38475451/06113133-40d6-4ef1-8006-eea008de31ad)

In both cases its same value of the color: RAL 1003 converted to hex #e5be01.
I change material programatically and I don't use any post processing.

Error from console
![image](https://github.com/google/model-viewer/assets/38475451/69018286-ea9d-45a3-b78e-23d85a987881)

Does anyone know why that happens?

**Comment by elalish** [comment]
Those look like pretty gnarly errors; I'm surprised you see a render at all. There's really no way we can know unless we have a live link to test on and a description of what device, OS, and browser (and versions) broke.

**Reply by piotrszpruta** [reply]
There's live link: https://ar.zcoderx.com/kreator/constanza.html
Information about device where it doesn't work:

> **Browser**
> Chrome 113.0.5672.127 (Firefox etc is the same)
> **Operating System**
> Windows 10 Pro 64-bit 22H2 19045.2965
> **CPU**
> Intel Core i3-2130 @ 3.40GHz
> Sandy Bridge 32nm Technology
> **RAM**
> 12,0GB
> **Motherboard**
> Dell Inc. 0NKW6Y (CPU 1)
> **Graphics**
> S24F350 (1920x1080@60Hz)
> Intel HD Graphics (Dell)
> **Storage**
> 119GB Hitachi HS-SSD-E100 128G (SATA (SSD))	31 °C
> **Audio**
> Realtek High Definition Audio

Please tell me if it's enough
P.S
Site is in Polish so please click on "Rozpocznij" and it will load view, on the left side you can choose colors.

**Reply by elalish** [reply]
Thanks for the info - it's working great here, and sadly I don't have a windows machine to try a repro with. @mrdoob Have you heard of any weird Windows 10 color problems? Best guess is this is some kind of GPU driver problem - make sure everything has been updated and beyond that, we may be out of luck.

**Reply by piotrszpruta** [reply]
I've tested and on windows 7 it's also problematic. I can send its hardware specification tomorrow (polish time)

**Reply by mrdoob** [reply]
@elalish I haven't 🤔

**Reply by piotrszpruta** [reply]
I think the problem is with some of the Intel GPUs. The problem randomly appears and I don't have an idea why 🫥

**Reply by piotrszpruta** [reply]
@elalish @mrdoob It's also happening on the https://modelviewer.dev/examples/augmentedreality/index.html, all models have darker textures or just pure black. Same error appears in console. All drivers are updated

**Reply by mrdoob** [reply]
> On mobile also, but i've checked on other 2 computers and textures appear I think darker?

What are the specs of that mobile device?

And the specs of the other two computers?

**Reply by piotrszpruta** [reply]
@mrdoob Mobile is working well, its Iphone 13 Pro and I've tested in on Xiaomi Redmi 8.
First PC with Windows 10:
```
Browser
Chrome 113.0.5672.127 (Firefox etc is the same)
Operating System
Windows 10 Pro 64-bit 22H2 19045.2965
CPU
Intel Core i3-2130 @ 3.40GHz
Sandy Bridge 32nm Technology
RAM
12,0GB
Motherboard
Dell Inc. 0NKW6Y (CPU 1)
Graphics
Intel HD Graphics (Dell)
```

Second PC with Windows 7:
```
Operating System
Windows 7 Professional 64-bit SP1
CPU
Intel Core i3 2120 @ 3.30GHz	42 °C
Sandy Bridge 32nm Technology
RAM
8,00GB Dual-Channel DDR3 @ 665MHz (9-9-9-24)
Motherboard
Dell Inc. 0J3C2F (CPU 1)
Graphics
Intel HD Graphics
Manufacturer	Intel
Model	HD Graphics
Device ID	8086-0102
Revision	A
Subvendor	Dell (1028)
Current Performance Level	Level 0
Current GPU Clock	847 MHz
Technology	32 nm
Driver version	9.17.10.3040
Count of performance levels : 1
Level 1 - "Perf Level 0"
GPU Clock 850 MHz
```

Third PC with Windows 10 has same specs as second one.
If you need any special information please tell me.

**Reply by mrdoob** [reply]
Seems like an issue with Intel GPUs then...

**Reply by piotrszpruta** [reply]
On UHD 770 GPU it works, on intel i7-12700

**Reply by piotrszpruta** [reply]
Intel i7-11700k | UHD 750 also works. I will try to search for more devices to test if it's required to do.

**Reply by piotrszpruta** [reply]
@mrdoob @elalish sorry for mentioning but is it caused by modelviewer? or three.js? It might will be resolved?

**Reply by mrdoob** [reply]
It's caused by Intel 🙂

**Reply by piotrszpruta** [reply]
I can possibly check if WebGL is created and if not return proper message to user?
